### PR TITLE
Search

### PR DESCRIPTION
**Pull request by natekuhns**

### CODE REVIEW

File: src/main/java/Main.java [hunk 1]
```diff
@@ -59,6 +59,7 @@ public static boolean is_valid_password(String password) throws IllegalArgumentE
 
     //account has a schedule instance that is worked on
     public static void populate_allcourses() throws IOException {
+        allcourses = new ArrayList<>();
```
Comment (Stuchellbj21) [inline src/main/java/Main.java:62]: What is the decision behind calling allcourses here and not in load_accounts?

Comment (natekuhns) [reply src/main/java/Main.java:62]: It makes sense to initialize allcourses when we're going to fill it rather than in some other place.... I wanted to be able to use populate_allcourses in tests and with '`allcourses = new ArrayList<>();`' in load_accounts you get an exception

File: src/main/java/Search.java [hunk 2]
```diff
@@ -36,15 +36,14 @@ public class Search {
...
     Search(String ss) {
         searchresults = new ArrayList<>();
-        search(ss);
         activefilters = new ArrayList<>();
+        search(ss);
     }
 
     Search(String ss, List<Filter> filters) {
         searchresults = new ArrayList<>();
-        search(ss);
```
Comment (Stuchellbj21) [inline src/main/java/Search.java:45]: Why the move of calling search?

File: src/main/java/Search.java [hunk 3]
```diff
@@ -131,18 +130,37 @@ public int get_weight(Course c) {
         for(String s : search_str_list) {
             if (c.get_id().contains(s)) {
                 //section
-                if(s.length() == 1 && Character.isAlphabetic(s.charAt(0))) w++;
+                if(s.length() == 1 && Character.isAlphabetic(s.charAt(0)) && c.getSection() == Character.toUpperCase(s.charAt(0))) w+=2;
                 //coursenum
-                else if(s.length() == 3 && Main.is_numeric(s)) w+=3;
+                else if(s.length() == 3 && Main.is_numeric(s)) w+=4;
```
Comment (Stuchellbj21) [inline src/main/java/Search.java:135]: Why the change to w in these elseifs?

**Review by Stuchellbj21 — APPROVED** [review]
